diff -ruN onyx-intl-booxsdk-3299a8a.org/CMakeLists.txt onyx-intl-booxsdk-3299a8a/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/CMakeLists.txt	2011-01-02 11:04:13.662350588 +0100
@@ -2,16 +2,13 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 2.8)
 
 # Additional cmake modules
-SET(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake_modules")
-
-# Naboo project.
-Project(SDK CXX C)
-ENABLE_TESTING()
+SET(CMAKE_MODULE_PATH "${ONYX_SDK_ROOT}/cmake_modules")
 
 # Check to use arm toolchain or not
 IF(BUILD_FOR_ARM)
     INCLUDE(arm_toolchain)
     USE_ARM_TOOLCHAIN()
+    link_directories("/opt/onyx/arm/lib")
 ENDIF(BUILD_FOR_ARM)
 
 IF(UNIX OR BUILD_FOR_ARM)
@@ -29,6 +26,10 @@
 ADD_DEFINITIONS(-DCONFIG_CTRL_IFACE_UNIX)
 ENDIF (WIN32)
 
+# Naboo project.
+Project(SDK CXX C)
+ENABLE_TESTING()
+
 INCLUDE(strict_warning)
 
 # Project include directories.
diff -ruN onyx-intl-booxsdk-3299a8a.org/code/examples/dictionary/CMakeLists.txt onyx-intl-booxsdk-3299a8a/code/examples/dictionary/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/code/examples/dictionary/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/code/examples/dictionary/CMakeLists.txt	2011-01-02 11:04:13.664347673 +0100
@@ -12,7 +12,10 @@
 add_definitions(-DQT_SHARED)
 add_definitions(-DQT_NO_DEBUG)
 
-add_library(my_dict_plugin SHARED ${DICT_SRCS})
-target_link_libraries(my_dict_plugin ${QT_LIBRARIES} ${ADD_LIB})
-
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(my_dict_plugin ${DICT_SRCS})
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(my_dict_plugin SHARED ${DICT_SRCS})
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
 
+target_link_libraries(my_dict_plugin ${QT_LIBRARIES} ${ADD_LIB})
diff -ruN onyx-intl-booxsdk-3299a8a.org/code/src/base/CMakeLists.txt onyx-intl-booxsdk-3299a8a/code/src/base/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/code/src/base/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/code/src/base/CMakeLists.txt	2011-01-02 11:04:13.667348784 +0100
@@ -5,10 +5,12 @@
   ${ONYXSDK_DIR}/include/onyx/base/tasks_handler.h
 )
 
-add_library(onyx_base ${MOC_SRCS}
-  base_model.cpp
-  tasks_handler.cpp
-)
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(onyx_base ${MOC_SRCS} base_model.cpp tasks_handler.cpp)
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(onyx_base SHARED ${MOC_SRCS} base_model.cpp tasks_handler.cpp)
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+
 target_link_libraries(onyx_base
   onyx_data # This is added for base_model. Perhaps base_model should be in onyx_data
 )
diff -ruN onyx-intl-booxsdk-3299a8a.org/code/src/CMakeLists.txt onyx-intl-booxsdk-3299a8a/code/src/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/code/src/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/code/src/CMakeLists.txt	2011-01-02 11:04:13.670348721 +0100
@@ -1,7 +1,11 @@
 enable_qt()
 
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(onyxapp application.cpp onyx_main.cpp)
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(onyxapp SHARED application.cpp onyx_main.cpp)
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
 
-add_library(onyxapp application.cpp onyx_main.cpp)
 target_link_libraries(onyxapp onyx_sys onyx_screen ${QT_LIBRARIES})
 strict_warning(onyxapp)
 
@@ -13,6 +17,6 @@
 add_subdirectory(sys)
 add_subdirectory(ui)
 add_subdirectory(wireless)
-add_subdirectory(sound)
-add_subdirectory(tts)
-add_subdirectory(dictionary)
+add_subdirectory(sound)
+add_subdirectory(tts)
+add_subdirectory(dictionary)
diff -ruN onyx-intl-booxsdk-3299a8a.org/code/src/cms/CMakeLists.txt onyx-intl-booxsdk-3299a8a/code/src/cms/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/code/src/cms/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/code/src/cms/CMakeLists.txt	2011-01-02 11:04:13.672348827 +0100
@@ -14,5 +14,10 @@
   user_db.cpp
   download_db.cpp)
 
-add_library(onyx_cms ${SRCS})
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(onyx_cms ${SRCS})
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(onyx_cms SHARED ${SRCS})
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+
 strict_warning(onyx_cms)
diff -ruN onyx-intl-booxsdk-3299a8a.org/code/src/data/CMakeLists.txt onyx-intl-booxsdk-3299a8a/code/src/data/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/code/src/data/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/code/src/data/CMakeLists.txt	2011-01-02 11:04:13.675349107 +0100
@@ -22,7 +22,7 @@
   ${ONYXSDK_DIR}/include/onyx/data/handwriting_functions_model.h
 )
 
-add_library(onyx_data
+SET(SRCS
   configuration.cpp
   bookmark.cpp
   reading_history.cpp
@@ -46,5 +46,12 @@
   content.cpp
   ${MOC_SRCS}
 )
+
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(onyx_data ${SRCS})
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(onyx_data SHARED ${SRCS})
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+
 target_link_libraries(onyx_data onyx_cms onyx_sys)
 strict_warning(onyx_data)
diff -ruN onyx-intl-booxsdk-3299a8a.org/code/src/dictionary/CMakeLists.txt onyx-intl-booxsdk-3299a8a/code/src/dictionary/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/code/src/dictionary/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/code/src/dictionary/CMakeLists.txt	2011-01-02 11:04:13.678347628 +0100
@@ -29,8 +29,12 @@
 SET(resource_srcs dictionary_images.qrc)
 QT4_ADD_RESOURCES(dictionary_ui_res ${resource_srcs})
 
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	ADD_LIBRARY(dictionary ${SRCS} ${MOC_SRCS} ${dictionary_ui_res})
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	ADD_LIBRARY(dictionary SHARED ${SRCS} ${MOC_SRCS} ${dictionary_ui_res})
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
 
-ADD_LIBRARY(dictionary ${SRCS} ${MOC_SRCS} ${dictionary_ui_res})
 TARGET_LINK_LIBRARIES(dictionary onyx_ui)
 
 ########### Unit Tests #####################
diff -ruN onyx-intl-booxsdk-3299a8a.org/code/src/screen/CMakeLists.txt onyx-intl-booxsdk-3299a8a/code/src/screen/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/code/src/screen/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/code/src/screen/CMakeLists.txt	2011-01-02 11:04:13.681348554 +0100
@@ -1,2 +1,7 @@
-add_library(onyx_screen STATIC screen_proxy.cpp)
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(onyx_screen STATIC screen_proxy.cpp)
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(onyx_screen SHARED screen_proxy.cpp)
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+
 install(TARGETS onyx_screen DESTINATION lib)
diff -ruN onyx-intl-booxsdk-3299a8a.org/code/src/sound/CMakeLists.txt onyx-intl-booxsdk-3299a8a/code/src/sound/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/code/src/sound/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/code/src/sound/CMakeLists.txt	2011-01-02 11:04:13.683345917 +0100
@@ -10,7 +10,11 @@
 SET(srcs ${srcs} ${hrds} ${MOC_SRCS})
 
 # Sound library.
-ADD_LIBRARY(sound ${srcs})
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	ADD_LIBRARY(sound ${srcs})
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	ADD_LIBRARY(sound SHARED ${srcs})
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
 
 # Unittest
 add_definitions(-DSAMPLE_ROOT="${CMAKE_SOURCE_DIR}/unittest/data")
diff -ruN onyx-intl-booxsdk-3299a8a.org/code/src/sys/CMakeLists.txt onyx-intl-booxsdk-3299a8a/code/src/sys/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/code/src/sys/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/code/src/sys/CMakeLists.txt	2011-01-02 11:04:13.686349886 +0100
@@ -13,7 +13,12 @@
 FILE(GLOB SRCS *.cpp)
 SET(SRCS ${HDRS} ${SRCS} ${MOC_SRCS})
 
-ADD_LIBRARY(onyx_sys ${SRCS})
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	ADD_LIBRARY(onyx_sys ${SRCS})
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	ADD_LIBRARY(onyx_sys SHARED ${SRCS})
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+
 strict_warning(onyx_sys)
 TARGET_LINK_LIBRARIES(onyx_sys onyx_wpa)
 
diff -ruN onyx-intl-booxsdk-3299a8a.org/code/src/tts/CMakeLists.txt onyx-intl-booxsdk-3299a8a/code/src/tts/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/code/src/tts/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/code/src/tts/CMakeLists.txt	2011-01-02 11:04:13.700348329 +0100
@@ -61,7 +61,11 @@
 SET(resource_srcs tts_images.qrc)
 QT4_ADD_RESOURCES(tts_ui_res ${resource_srcs})
 
-ADD_LIBRARY(tts ${SRCS} ${MOC_SRCS} ${tts_ui_res})
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	ADD_LIBRARY(tts ${SRCS} ${MOC_SRCS} ${tts_ui_res})
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	ADD_LIBRARY(tts SHARED ${SRCS} ${MOC_SRCS} ${tts_ui_res})
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
 
 TARGET_LINK_LIBRARIES(tts sound  onyx_ui
    ${QT_LIBRARIES}
diff -ruN onyx-intl-booxsdk-3299a8a.org/code/src/wireless/CMakeLists.txt onyx-intl-booxsdk-3299a8a/code/src/wireless/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/code/src/wireless/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/code/src/wireless/CMakeLists.txt	2011-01-02 11:04:13.702347946 +0100
@@ -14,7 +14,13 @@
 QT4_ADD_RESOURCES(WIFI_RES ${resource_srcs})
 
 QT4_WRAP_CPP(MOC_SRCS ${HDRS})
-ADD_LIBRARY(onyx_wireless ${SRCS} ${MOC_SRCS} ${WIFI_RES})
+
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	ADD_LIBRARY(onyx_wireless ${SRCS} ${MOC_SRCS} ${WIFI_RES})
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	ADD_LIBRARY(onyx_wireless SHARED ${SRCS} ${MOC_SRCS} ${WIFI_RES})
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+
 strict_warning(onyx_wireless)
 INSTALL(TARGETS onyx_wireless DESTINATION lib)
 
diff -ruN onyx-intl-booxsdk-3299a8a.org/code/src/wpa/CMakeLists.txt onyx-intl-booxsdk-3299a8a/code/src/wpa/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/code/src/wpa/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/code/src/wpa/CMakeLists.txt	2011-01-02 11:04:13.709348465 +0100
@@ -6,6 +6,10 @@
     SET(SRC wpa_ctrl.c os_unix.c)
 ENDIF (WIN32)
 
-ADD_LIBRARY(onyx_wpa ${SRC})
-strict_warning(onyx_wpa)
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	ADD_LIBRARY(onyx_wpa ${SRC})
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	ADD_LIBRARY(onyx_wpa SHARED ${SRC})
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
 
+strict_warning(onyx_wpa)
diff -ruN onyx-intl-booxsdk-3299a8a.org/code/tests/CMakeLists.txt onyx-intl-booxsdk-3299a8a/code/tests/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/code/tests/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/code/tests/CMakeLists.txt	2011-01-02 11:04:13.714347803 +0100
@@ -30,11 +30,11 @@
   WILL_FAIL TRUE)
 
 onyx_test(conf_unittest conf_unittest.cpp configuration.cpp)
-TARGET_LINK_LIBRARIES(conf_unittest onyx_cms onyx_data ${QT_LIBRARIES})
+TARGET_LINK_LIBRARIES(conf_unittest onyx_cms onyx_data onyx_screen ${QT_LIBRARIES})
 ADD_TEST(BookmarkUnittest ${TEST_OUTPUT_PATH}/conf_unittest)
 
 ADD_EXECUTABLE(bookmark_unittest bookmark_unittest.cpp)
-TARGET_LINK_LIBRARIES(bookmark_unittest onyx_cms gtest_main onyx_data ${QT_LIBRARIES})
+TARGET_LINK_LIBRARIES(bookmark_unittest onyx_cms gtest_main onyx_data onyx_screen ${QT_LIBRARIES})
 MAYBE_LINK_TCMALLOC(bookmark_unittest)
 SET_TARGET_PROPERTIES(bookmark_unittest PROPERTIES  RUNTIME_OUTPUT_DIRECTORY ${TEST_OUTPUT_PATH})
 ADD_TEST(BookmarkUnittest ${TEST_OUTPUT_PATH}/bookmark_unittest)
diff -ruN onyx-intl-booxsdk-3299a8a.org/code/tests/sys/CMakeLists.txt onyx-intl-booxsdk-3299a8a/code/tests/sys/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/code/tests/sys/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/code/tests/sys/CMakeLists.txt	2011-01-02 12:16:41.703345378 +0100
@@ -6,10 +6,10 @@
 
 
 ADD_EXECUTABLE(wifi_manager_unittest wifi_manager_test.cpp)
-TARGET_LINK_LIBRARIES(wifi_manager_unittest onyx_cms onyx_data onyx_sys ${QT_LIBRARIES})
+TARGET_LINK_LIBRARIES(wifi_manager_unittest onyx_cms onyx_data onyx_sys onyx_screen ${QT_LIBRARIES})
 SET_TARGET_PROPERTIES(wifi_manager_unittest PROPERTIES  RUNTIME_OUTPUT_DIRECTORY ${TEST_OUTPUT_PATH})
 
 ADD_EXECUTABLE(generate_default_peers generate_default_peers.cpp)
-TARGET_LINK_LIBRARIES(generate_default_peers onyx_cms onyx_data onyx_sys ${QT_LIBRARIES})
+TARGET_LINK_LIBRARIES(generate_default_peers onyx_cms onyx_data onyx_sys onyx_screen ${QT_LIBRARIES})
 
 
diff -ruN onyx-intl-booxsdk-3299a8a.org/third_party/gmock/CMakeLists.txt onyx-intl-booxsdk-3299a8a/third_party/gmock/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/third_party/gmock/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/third_party/gmock/CMakeLists.txt	2011-01-02 11:04:13.722344009 +0100
@@ -1,7 +1,11 @@
 include_directories(. include)
 
-add_library(gmock
-  src/gmock-all.cc)
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(gmock src/gmock-all.cc)
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(gmock SHARED src/gmock-all.cc)
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+
 set_target_properties(gmock
   PROPERTIES
   COMPILE_FLAGS "-DPATH_MAX=4096")
diff -ruN onyx-intl-booxsdk-3299a8a.org/third_party/gtest/CMakeLists.txt onyx-intl-booxsdk-3299a8a/third_party/gtest/CMakeLists.txt
--- onyx-intl-booxsdk-3299a8a.org/third_party/gtest/CMakeLists.txt	2011-01-01 10:22:37.000000000 +0100
+++ onyx-intl-booxsdk-3299a8a/third_party/gtest/CMakeLists.txt	2011-01-02 11:04:13.728346554 +0100
@@ -1,10 +1,19 @@
 include_directories(.)
 
-add_library(gtest
-  src/gtest-all.cc)
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(gtest src/gtest-all.cc)
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(gtest SHARED src/gtest-all.cc)
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+
 set_target_properties(gtest
   PROPERTIES
   COMPILE_FLAGS "-DPATH_MAX=4096")
 
-add_library(gtest_main src/gtest_main.cc)
+IF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(gtest_main src/gtest_main.cc)
+ELSE (WIN32 OR ONYX_BUILD_STATIC_LIB)
+	add_library(gtest_main SHARED src/gtest_main.cc)
+ENDIF (WIN32 OR ONYX_BUILD_STATIC_LIB)
+
 target_link_libraries(gtest_main gtest)
